cmd/metal-api/internal/service/machine-service.go

```diff
@@ -958,16 +960,39 @@ func allocateMachine(logger *zap.SugaredLogger, ds *datastore.RethinkStore, ipam
 		return nil, err
 	}
 
+	var fsl *metal.FilesystemLayout
+	if allocationSpec.FilesystemLayoutID == "" {
```

Comment:
I think making this a pointer and optional in the spec would be clearer because how does the consumer of the API know that an empty string does this implicit find?

Reply:
Done